### ISSUE TITLE
[Feature] See user badge when reply

### ISSUE THREAD

**Issue by PeopleInside** [flarum/core#1765]
Feature approved by Toby

I would suggest to add the badge when user post or reply also where the red row show. This alow who are posting to remember he his posting from a special profile... just remember is not a reply from a "normal" profile as actually the forum show.

This is useful also when a new post is created. Will be nice because who are writting a post can see they are writting as badged user.

![00](https://user-images.githubusercontent.com/5006150/54914813-261f0800-4ef6-11e9-8a99-ef3a7eb85144.png)
![01](https://user-images.githubusercontent.com/5006150/54914819-2b7c5280-4ef6-11e9-9a8f-cbb6db8e318a.png)

**Justify why this feature belongs in Flarum's core, rather than in a third-party extension**
Consider who this change will be useful to – most Flarum forums, or just a few?

This is part of the community why should be an extension?

**Comment by PeopleInside** [comment]
up

**Comment by PeopleInside** [comment]
up

**Comment by askvortsov1** [comment]
This is pretty nice as a good first issue

**Comment by askvortsov1** [comment]
Bump. Essentially for the preview, PostUser would need to be added to `PostPreview` For composer I'm not as sure we need the badges there (post preview ensures consistency, but in composer it would add clutter IMO).

**Comment by PeopleInside** [comment]
Hi, I see how the new feature work and don't like fully.
You choose to show the badge in the post reply but this will not work if there is no post preview for example when you create a new post.

Why did not include also the badge in the composer window?

![SharedScreenshot](https://user-images.githubusercontent.com/5006150/107615862-39076680-6c4d-11eb-9a49-3223b99d3a33.jpg)
![SharedScreenshot 1](https://user-images.githubusercontent.com/5006150/107615873-3b69c080-6c4d-11eb-8bb1-36914bcafb57.jpg)

**Comment by askvortsov1** [comment]
IMO this is about consistency. Not showing the badges with the post previews was not ideal because the post preview looks exactly like other posts. But the composer doesn't have that restriction. Without the factor of consistency, showing badges on the composer doesn't really make sense to me, since the user is already aware of their groups, and it would add clutter to a UI element that would preferably be kept simple.

**Comment by PeopleInside** [comment]
Not showing the badge when posting is the real initial issue.
Maybe I think is more login show the bade in the composer respect showing in preview also if I think should be showed everywhere as is correct to me show in preview and in the composer.

The fix made here is partially and will "solve" or be useful only for replies and not for a new topic creation.
Avatar of the user are showed in post, preview and composer so why badge should be not showed in the same way?

**Comment by SychO9** [comment]
I agree with @askvortsov1, the avatar in the composer is a nice design choice, it's not there to really show information about the logged in user, I don't see why we would need to add anything else, it really is just more clutter to the composer UI.

And as mentioned above, the composer element is not a post element, so there is no need in trying to make it look consistent anyway.

**Comment by PeopleInside** [comment]
> 
> 
> I agree with @askvortsov1, the avatar in the composer is a nice design choice, it's not there to really show information about the logged in user, I don't see why we would need to add anything else, it really is just more clutter to the composer UI.
> 
> And as mentioned above, the composer element is not a post element, so there is no need in trying to make it look consistent anyway.

This maybe consider the fact you know your rule as admin if you own your own forum.
Will be not the same in a community where users can be upgraded.

In any case I saw my opinion: this fix is not complete is partial and I feel not correct to show only somewhere the user group and not show in other place where you figure out as regular user. This is just my opinion, you will decide maybe and this will be closed but I don't agree.

Again the fix done is partially and solve only the question when you reply if you are seeing the preview.
Show no badge in the composer looks like to be an error for me as the system are showed your profile as regular user and this is not a right information.

**Comment by matteocontrini** [comment]
> In the following screen you can see no indication user that are creating a topic is an admin:

How can an admin not know that he/she is an admin? I think the purpose of group badges is to let _other_ users know to which groups a user belongs to, because they might forget or not know. A user may be upgraded, yes, but surely they won't discover they've been upgraded while posting a new discussion... :)

**Comment by PeopleInside** [comment]
Great so I suppose this seems not to be liked.
Maybe when close this issue without any other fix can you tell me how I can add this? Editing composerbody.js?

I want try to implement on my side if will be not implemented here.

**Comment by askvortsov1** [comment]
I think we can keep this issue open for now, and as a compromise, refactor Composer Body to use an item list so that extensions can modify this if they wish?

**Comment by PeopleInside** [comment]
I will wait for a solution.
I'm asking what sense to add only the badge to the preview were you see already your badges in the existent post.

Was having much more sense add it to the composer where if you create new discussion you don't see the badge...

Anyway will wait to know how can I add badges to the composer.

A lot of code or just a simple line as for the fix done in the preview? :)

I also don't know how to create an extension for solve this.

Il giorno 12 feb 2021, 01:18, alle ore 01:18, Alexander Skvortsov <notifications@github.com> ha scritto:

**Comment by PeopleInside** [comment]
Not stale :(
Hope to understand how I can show badge also in composer.
